src/lib/y2users/password.rb

```diff
@@ -93,5 +103,46 @@ def ==(other)
       # do not compare configuration to allow comparison between different configs
       ATTRS.all? { |a| public_send(a) == other.public_send(a) }
     end
+
+    # Represents password value. It specific type is defined as subclass and can be queried
+    class Value
+      include Yast2::SecretAttributes
+
+      secret_attr :content
+
+      def initialize(content)
+        self.content = content
+      end
+
+      def plain?
+        false
+      end
+
+      def encrypted?
+        false
+      end
+    end
+
+    # Represents encrypted password value or its specific types like disabled of locked logging.
+    class EncryptedValue < Value
+      def encrypted?
+        true
+      end
+
+      def locked?
+        content.start_with?("!$")
+      end
+
+      def disabled?
+        ["*", "!"].include?(content)
```

Comment:
Not sure if this is because 

> If the password field contains some string that is not a valid result of crypt(3), for instance ! or *, the user will not be able to use a unix password to log in (but the user may log in the system by other means). — _[man shadow(5)](https://linux.die.net/man/5/shadow)_

If so, there are other characters that should be included, 

>  It will be entirely printable ASCII, and will not contain **whitespace** or the characters ‘**:**’, ‘**;**’, ‘**\***’, ‘**!**’, or ‘**\\**’. — _[man crypt(3)](https://linux.die.net/man/3/crypt)_

Reply:
:thinking: :thinking: :thinking: 

I see now that you're honoring what is stated in the class [documentation](https://github.com/yast/yast-users/blob/9be29d36f9d6619082856f8942c9abd1b279b5de/src/lib/y2users/password.rb#L29-L32)

I guess it's a _common_ practice, since I was not able to found an official documentation about it beyond the people using an invalid character  in at the beginning, but not replacing it completely.

Moreover, `usermod`and `passwd` prefix the value with an exclamation mark to lock the password, which is checked by the `lock?` method.

Reply:
Me neither and what is worry that it often depends on implementation. Encrypted password start with "$<hash encoded type like 4,5,x or 2y>$" so anything else beside "!<encrypted password>" is probably disabled. For now  I think that * and ! covers majority of usage at least on suse.